src/Wallabag/CoreBundle/Controller/EntryController.php

```diff
@@ -270,6 +270,7 @@ public function redirectRandomEntryAction($type = 'all')
     public function viewAction(Entry $entry)
     {
         $this->checkUserAction($entry);
+        $entry->setLanguage($entry->getHTMLLanguage());
```

Comment:
I'm not a big fan of overriding the already saved data with something else only when rendering the entry.

Instead, you should keep your `getHTMLLanguage` and use it directly in the twig using `entry.html_language` (I'm not sure about the `_`)